### DISCUSSION TITLE
Did you stop updating the Firefox version ?

### DISCUSSION THREAD

**Discussion by I-I-IT** [uBlockOrigin/uBOL-home#231] [Q&A]
I know There was a spat between @gorhill and Mozilla AMO. But I believe you should still self-distribute it here on Github. Because, if not, many people will miss out on updates for filter list.  Some people focused about security might go to Chromium if uBol is now unmaintained on Firefox. Thank you

**Comment by hj-collab** [comment]
@gorhill Same thoughts as OP. Please revive the uBol. 🙏 uBol is godsent for people with higher focus on security and low powered devices.

**Reply by garry-ut99** [reply]
Here is a background of how uBOL for Firefox appeared (for these who don't know):

- https://github.com/uBlockOrigin/uBOL-home/discussions/28

> I-I-IT : I know There was a spat between

Yes, and here is a background of how uBOL for Firefox disappeared (for these who don't know):
- https://github.com/uBlockOrigin/uBOL-home/issues/197

> hj-collab

Last time in your 2 issues you were using uBOL for Chromium: https://github.com/uBlockOrigin/uBOL-home/issues/23 + https://github.com/uBlockOrigin/uBOL-home/issues/62, have you switched to Firefox recently or you use it on both Chromium & Firefox.

**Reply by hj-collab** [reply]
> Last time in your 2 issues you were using uBOL for Chromium: #23 + #62, have you switched to Firefox recently or you use it on both Chromium & Firefox.

Yes, I've recently switched to Firefox completely and now use Chromium solely for testing.

**Comment by alsoLut** [comment]
if the burden is to much for @gorhill surely someone else could pick this minor task, right?  The ideea of building a open source project is to share it with as many people as possible.

**Reply by MasterKia** [reply]
> surely someone else could pick this minor task

Please feel free to do so.

**Reply by alsoLut** [reply]
Do you really want hundreds of clones using the uBlock name? This should be handled under the original developer's name.

**Reply by stephenhawk8054** [reply]
It's a [GPLv3 License](https://github.com/gorhill/uBlock/blob/master/LICENSE.txt) project. Anyone can fork and publish it as long as they comply with the license. Dev could not prevent the "clones", and he's not obliged to do what he does not want to. Preventing malicious add-ons should be the jobs of AMO, not him.

**Reply by alsoLut** [reply]
> It's a [GPLv3 License](https://github.com/gorhill/uBlock/blob/master/LICENSE.txt) project. Anyone can fork and publish it as long as they comply with the license. Dev could not prevent the "clones", and he's not obliged to do what he does not want to. Preventing malicious add-ons should be the jobs of AMO, not him.

I did not imply anything you mentioned above.

**Reply by stephenhawk8054** [reply]
Then the clones are not the issue, as long as they comply with the license.

**Reply by alsoLut** [reply]
> Then the clones are not the issue, as long as they comply with the license.

The idea is to discourage clones by offering an official extension with numerous positive reviews.

**Reply by stephenhawk8054** [reply]
Dev is not obliged to discourage them if they comply the license correctly. If anything, that just forces him to put more work and time to something that's already causing many frictions. When uBO was not on Safari or Legacy Edge, somebody else volunteered to pick up the work with the stores and communicated back clearly, reducing dev's work and time. You can do it too, anyone else can do it too.

**Comment by gorhill** [comment]
> surely someone else could pick this minor task

What is needed for this to happen is to have volunteers contributing by:

- Building uBO Lite locally
- Using the locally built version to verify that all work as expected for each release
    - Verify that at first install the extension first-install page opens as expected
    - Verify that first-install page doesn't open on subsequent launch
    - Verify that all filtering modes work as expected on many sites, with different set of rulesets enabled
    - Verify that admin deployment options work as expected
     - etc.

On both Firefox and Firefox for Android.

If I have guarantees that it is well tested, I would feel confident to again publish a version on AMO.

**Reply by alsoLut** [reply]
You might want to make a more public call to action outlining these expected steps. I'm confident we can gather volunteers.

**Reply by I-I-IT** [reply]
I could help other volunteers to do this testing, but yeah I don't want to do this on my own, especially every week.

**Reply by alsoLut** [reply]
> Verify that all filtering modes work as expected on many sites, with different set of rulesets enabled 


can you please expand on this step?

**Reply by kuro68k** [reply]
All this could be done with Selenium, but it will need an expert in that application to set it up and maintain it.

The only other comment I would make is that I'd be happy to use a much less well tested build, simply because there is no viable alternative for Android. I'd rather accept some breakage than the crippling battery drain that comes with full uBlock Origin on Firefox for Android.

So to that end an unofficial build that auto-updates when uBOL is released for Chromium browsers would be welcome. I'd volunteer but I'm not familiar with any of this stuff, I'm an embedded C developer. Anyone with knowledge of this care to comment on what is required? Trigger on release, fetch source, patch the update URL, build, push release to Github?

**Reply by gorhill** [reply]
> crippling battery drain that comes with full uBlock Origin on Firefox for Android

Do you have objective evidence to support such statement? (that is, objective evidence for browsing with vs. without uBO) There is already too much bad technical information out there arising from bad presumptions about how things work, and the best way to avoid this is to stick to objective evidence. Firefox as an excellent profiler to sort out the amount of work done by the CPU with and without a given extension, there is no need to presume anything.

The DNR implementation in Firefox [is JavaScript-based](https://searchfox.org/mozilla-central/rev/b655023a1978bcad8cc2fb038b89a058b45e91a7/toolkit/components/extensions/ExtensionDNR.sys.mjs), and is far less optimized than uBO ([related bugzilla issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1745768)). The DNR engine in Firefox currently iterates through all rules in all enabled rulesets ([code](https://searchfox.org/mozilla-central/rev/b655023a1978bcad8cc2fb038b89a058b45e91a7/toolkit/components/extensions/ExtensionDNR.sys.mjs#1763)), whereas only filters which have a good chance of matching are visited in uBO (see <https://x.com/gorhill/status/1843293047020507470>).

Additionally, DNR matching algorithm intrinsically requires more work since it must find all matching filters (in order to compare their priority), whereas uBO's matching algorithm bail out after the first match.

**Reply by kuro68k** [reply]
I don't have objective evidence, no. I can try it again but at least on my device (Pixel 8 Pro) there is a very significant difference between the battery consumption when browsing with Chrome or Firefox with uBOL, and Firefox with uBO. The difference is large, something like 5x worse in terms of % lost per hour when browsing.

If there is some way to objectively test it, I'll see if I can try it. While I'm sure you are right about what Firefox says in terms of CPU profiling, it could be something specific to this family of devices (it was the same on the Pixel 5 and 7 Pro, the 5 used a Qualcomm SoC and the 7 and 8 use Google's own silicon). Excessive CPU wake-ups due to the interaction of some timings or something.

I will retry today and report back.

**Reply by gorhill** [reply]
There are many factors that must be considered, like which websites were visited and for how long, how are the blockers configured (they should be as similarly configured as possible, which other extensions were present, and so on

**Reply by I-I-IT** [reply]
>If there is some way to objectively test it, I'll see if I can try it. While I'm sure you are right about what Firefox says in terms of CPU profiling, it could be something specific to this family of devices (it was the same on the Pixel 5 and 7 Pro, the 5 used a Qualcomm SoC and the 7 and 8 use Google's own silicon)


Kiwi Browser and Microsoft Edge have Android support for extensions. It would help to compare within the same browser.

**Reply by kuro68k** [reply]
> Kiwi Browser and Microsoft Edge have Android support for extensions. It would help to compare within the same browser.

Both are based on Chromium though, and the issue is likely related to Firefox. Chromium has a more efficient filtering engine for Manifest V3 extensions than Firefox does.

I have been testing it out, and I haven't checked the changelog since earlier this year when I last tested it, but it does seem to be substantially better now. I'm trying to come up with a good test plan, but it isn't easy. For the time being I've switched back to uBO for daily use.

**Reply by garry-ut99** [reply]
> The DNR implementation in Firefox [is JavaScript-based](https://searchfox.org/mozilla-central/rev/b655023a1978bcad8cc2fb038b89a058b45e91a7/toolkit/components/extensions/ExtensionDNR.sys.mjs), and is far less optimized than uBO ([related bugzilla issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1745768)). (...)

True, here also an additional link (which further leads to a Wiki link): 
- https://github.com/uBlockOrigin/uBOL-home/discussions/110

**Reply by kuro68k** [reply]
I did the best tests I could without automation, and I would say that whatever the issue was it is now fixed. Any difference in energy performance is within the margin of error. I saved an old nightly from a year ago and re-confirmed the issue there, so whatever it is has been fixed between then and now.

**Comment by gorhill** [comment]
There is a self-distributed version for Firefox, see [README](https://github.com/uBlockOrigin/uBOL-home?tab=readme-ov-file#ubo-lite)

**Reply by VadosG** [reply]
Will the extension for Firefox **self-update** to new versions?

**Reply by gorhill** [reply]
Yes, it should.
